UM/Qt/ListModel.py

```diff
@@ -161,14 +161,19 @@ def setProperty(self, index: int, property: str, value: Any) -> None:
         self._items[index][property] = value
         self.dataChanged.emit(self.index(index, 0), self.index(index, 0))
 
-    def sort(self, fun: Callable[[Any], float]) -> None:
+    def sort(self, fun: Callable[[Any], float], key: Optional[str] = None, reverse = False) -> None:
         """Sort the list.
 
         :param fun: The callable to use for determining the sort key.
+        :param key: Use the sorting function on the underlying data
+        :param reverse: reverse the sorted results
         """
 
         self.beginResetModel()
-        self._items.sort(key = fun)
+        if key:
+            self._items = [{key: item} for item in sorted([k[key] for k in self._items], key = fun, reverse = reverse)]
```

Comment:
I'm not sure how I feel about data being changed (format, not just positionally altered) by a sorting function. Calling this twice (each time) with a different key will result in an error for example., since it won't be able to find `k[key]` the 2nd time around, since that will be in `k[<key used first time>][key]` instead then.

Reply:
Yes you are indeed correct. Will change this